In [13]:
import pandas as pd
import numpy as np
import os
import glob
import matplotlib.pyplot as plt
import seaborn as sns

print("Libraries imported successfully!")

Libraries imported successfully!


In [15]:
data_path = r'C:\Users\PC\Desktop\News_Sentiment\data\raw\yfinance\Data'
#glob - Find all files matching a specified pattern
#os- provides a way to interact with computer's file system
# Find all CSV files in that folder
stock_files = glob.glob(f'{data_path}\\*.csv')

# Load and combine all stock files
stock_data = [] #at this point the stcok_data is a list of separate Dataframes

for file in stock_files:
    ticker = os.path.basename(file).replace('.csv', '') #extract the filename from full file path and remove the .csv
    df = pd.read_csv(file)
    df['Ticker'] = ticker
    stock_data.append(df)
    print(f"Loaded {ticker}: {len(df)} rows")

# but here we are combining all the separate Dataframes from stock_data in to one big dataframe
df_all_stocks = pd.concat(stock_data, ignore_index=True)

print("Combined DataFrames")
df.head()

Loaded AAPL: 3774 rows
Loaded AMZN: 3774 rows
Loaded GOOG: 3774 rows
Loaded META: 2923 rows
Loaded NVDA: 3774 rows
Combined DataFrames


,Date,Close,High,Low,Open,Volume,Ticker
0,2009-01-02,0.199652,0.201027,0.184294,0.184982,497124000,NVDA
1,2009-01-05,0.203319,0.207904,0.195984,0.197360,705736000,NVDA
2,2009-01-06,0.210196,0.216156,0.204695,0.209279,657904000,NVDA
3,2009-01-07,0.197589,0.205382,0.190483,0.205382,870096000,NVDA
4,2009-01-08,0.192546,0.195067,0.180626,0.195067,1014496000,NVDA


## Task 2: Loading and Combining Stock Data

### Data Source

The stock price data was sourced from multiple CSV files, one per stock ticker:

| Ticker | Company | Rows |
|--------|---------|------|
| AAPL | Apple Inc. | 3,774 |
| AMZN | Amazon.com Inc. | 3,774 |
| GOOG | Alphabet Inc. | 3,774 |
| META | Meta Platforms Inc. | 2,923 |
| NVDA | NVIDIA Corporation | 3,774 |

### Loading Process

The following steps were performed to load and combine the data:

#### 1. File Discovery
```python
stock_files = glob.glob(f'{data_path}\\*.csv')
```
#### 2. Individual file loading
For each CSV file found, the ticker name is extracted, loaded to pandas DataFrame and Ticker column is added
#### 3. Combinig DataFrames
stack all individual Dataframes from step 2 to a single big DataFrame and reset row index value so there won't be duplicated index value

In [17]:
df.columns

Index(['Date', 'Close', 'High', 'Low', 'Open', 'Volume', 'Ticker'], dtype='str')

In [18]:
print(df.isnull().sum())

Date      0
Close     0
High      0
Low       0
Open      0
Volume    0
Ticker    0
dtype: int64


The above result shows that there are no missing values for each row

In [19]:
print(df.duplicated().sum())

0


The above result shows there are no duplicated result in the given dataset 